utils.py

```diff
@@ -4,38 +4,56 @@
 from io import BytesIO
 from zipfile import ZipFile, ZIP_DEFLATED
 
-
-def get_zip(module_name, env):
-    output = BytesIO()
-    zf = ZipFile(output, "w", ZIP_DEFLATED)
-    external_modules = get_external_dependencies('industry/' + module_name, env)
-    dirs = ['industry/' + module_name] + ['industry/' + module for module in external_modules]
-    for dir in dirs:
-        for dirname, _, files in os.walk(dir):
-            zip_dirname = os.sep.join(dirname.split(os.sep)[1:])
-            zf.write(dirname, zip_dirname)
-            for filename in files:
-                zf.write(os.path.join(dirname, filename), os.path.join(zip_dirname, filename))
-    zf.close()
-    return output
-
-
-def get_dependencies(dir):
-    manifest_file = dir + '/__manifest__.py'
-    if not os.path.exists(manifest_file):
-        raise FileNotFoundError(f"Manifest file not found: {manifest_file}")
-    with open(manifest_file, encoding="utf-8") as f:
-        manifest = ast.literal_eval(f.read())
-    return set(manifest.get('depends', ['base']))
-
-
-def get_external_dependencies(dir, env):
-    dependencies = get_dependencies(dir)
-    known_modules = env['ir.module.module'].search([]).mapped('name')
-    return dependencies.difference(known_modules)
-
-
-def install_internal_dependencies(dir, env):
-    dependencies = get_dependencies(dir)
-    modules = env['ir.module.module'].search([('name', 'in', dependencies)])
-    return modules.button_immediate_install()
+class IndustryUtils:
+    def __init__(self, industry_path: str):
+        self.industry_path = industry_path
+
+
+    def get_zip(self, module_name: str):
+        output = BytesIO()
+        zf = ZipFile(output, "w", ZIP_DEFLATED)
+        industry_modules = self.get_industry_dependencies(module_name)
+        dirs = [self.industry_path + module for module in industry_modules]
+        for dir in dirs:
+            for dirname, _, files in os.walk(dir):
+                zip_dirname = os.sep.join(dirname.split(os.sep)[1:])
+                zf.write(dirname, zip_dirname)
+                for filename in files:
+                    zf.write(os.path.join(dirname, filename), os.path.join(zip_dirname, filename))
+        zf.close()
+        return output
+
+
+    def get_manifest(self, industry: str):
+        return self.industry_path + industry + '/__manifest__.py'
+
+
+    def is_industry(self, module: str):
+        return os.path.exists(self.get_manifest(module))
+
+
+    def get_dependencies(self, module_name: str):
+        industries = {module_name}
+        other_dep = set()
+        to_check = [module_name]
+        while to_check:
+            manifest_file = self.get_manifest(to_check.pop())
+            with open(manifest_file, encoding="utf-8") as f:
+                manifest = ast.literal_eval(f.read())
+            for dep in manifest.get('depends'):
```

Comment:
Maybe keep the fallback?
```suggestion
            for dep in manifest.get('depends', ['base']):
```